msfvenom

```diff
@@ -69,81 +69,69 @@ def parse_args(args)
   opt.separator('')
   opt.separator('Options:')
 
+  opt.on('-l', '--list            <type>', Array, 'List all module for [type]. Types are: payloads, encoders, nops, platforms, encrypt, formats, all') do |l|
+    if l.nil? or l.empty?
```

Comment:
`if l.to_s.empty?` would work here. Add `strip` or use `blank?` (Active Support) if you don't want blanks. You should also use `||` instead of `or`.

Reply:
`if l.to_s.empty?` I agree

`Add strip or use blank? (Active Support) if you don't want blanks` I don't understand.

`You should also use || instead of or.` I agree

Reply:
http://guides.rubyonrails.org/active_support_core_extensions.html#blank-questionmark-and-present-questionmark

`l.to_s.strip.empty?` is close enough for your purposes but rather ugly overall. I'm iffy on furthering a dependency on Active Support, but `blank?` is useful. They should add it to core. ;)